src/components/DatasetDownloadOptionsForm.js

```diff
@@ -0,0 +1,413 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import classnames from 'classnames';
+import Link from 'next/link';
+import { Formik, Form, Field } from 'formik';
+import { IoIosArrowUp, IoIosArrowDown } from 'react-icons/io';
+import * as yup from 'yup';
+
+import Alert from './Alert';
+import Button from './Button';
+import Checkbox from './Checkbox';
+import Dropdown from './Dropdown';
+import Error from './Error';
+import HelpIcon from './HelpIconLink';
+
+import { useLocalStorage } from '../common/hooks';
+import { formatSentenceCase, subscribeToMailingList } from '../common/helpers';
+import { ServerError } from '../common/errors';
+import { createToken } from '../state/token';
+import { createDataSet, updateDataSetOptions } from '../api/dataSet';
+
+// This is the mechanism for propagating properties to child components
+// TODO: Any option should be overridable at the child component level
+const DatasetDownloadOptionsContext = React.createContext();
+
+// If you do not add children this is what will be populated
+// Notice no EMAIL
+const DefaultOptions = () => {
+  return (
+    <DatasetDownloadOptionsContext.Consumer>
```

Comment:
Is it better to use the hook version of this [useContext](https://reactjs.org/docs/hooks-reference.html#usecontext)?

Reply:
I like this better, good look!